Customer forms

xidoCohenx opened the pull request.

xidoCohenx committed:
adding the addCustomerForm component

xidoCohenx committed:
customerForm validators addition

xidoCohenx committed:
added valuesImplementation

and using it onSubmit

xidoCohenx committed:
changed updateCustomerDetails to accept customer value

xidoCohenx committed:
fixed the validators and

added the patternvalidator
also added the f Short-cut
and changed the navigation to customerHome

OrShemesh5020 commented on src/app/components/add-customer-form/add-customer-form.component.ts:
```diff
@@ -10,60 +17,70 @@ import { Component, OnInit } from '@angular/core';
-
-    }
+    // and here, navigate to customer -HOme.
+    this.router.navigate(['customer/admin-home']);
```
the right navigation is customerHome (this.router.navigate(['customerHome']);)

xidoCohenx committed:
completed the add-customer-form template

xidoCohenx committed:
fixed the navigation

added validation at the start of onSubmit

xidoCohenx committed:
changed all customer's parameters

to be optionally given

xidoCohenx committed:
added navigation to addCustomerForm

in app routes

xidoCohenx committed:
added router  to sign up's constructor

xidoCohenx committed:
added the addCustomerForm's decorations

and shaping classes

xidoCohenx committed:
added the updateCustomerForm component

xidoCohenx committed:
fixed admin navigation to form

xidoCohenx committed:
removed loadCustomers becuase

 the navigate will take care of that
and took out the variables from the template

xidoCohenx committed:
fixed input types to text

and added admin/Customer type
verification.

xidoCohenx committed:
cleaned addCustomer and

added navigation to log out

xidoCohenx committed:
update-customer-form template first Draft

xidoCohenx committed:
added valuesImplementation

and validation check to ngOnInit

xidoCohenx committed:
added route to update-customer-form.component

xidoCohenx committed:
change the values' order so that ID  is first

xidoCohenx committed:
added updateDetails button

and changed customer model's id parameter to be first

xidoCohenx committed:
added the service and routers

to the constructor and
added navigation to updateCustomerForm

xidoCohenx committed:
moving ng on init to the arrowFunction of getCustomerDetails()

xidoCohenx committed:
added a comment to a problem that i found in the addCustomerForm component

xidoCohenx committed:
added the updateCustomerForm css

xidoCohenx committed:
added implementation to the form's different controls attribute to fix "FormGroup instance missing" error

OrShemesh5020 commented on src/app/app.routes.ts:
```diff
@@ -16,6 +17,7 @@ export const routes: Routes = [
   { path: 'companyHome', component: CompanyHomeComponent },
   { path: 'customerHome', component: CustomerHomeComponent },
   { path: 'addCustomerForm', component: AddCustomerFormComponent },
+  { path: 'updateCustomerForm', component: UpdateCustomerFormComponent},
```
change to 'updateCustomerForm/id'
because you want to transfer the customer's id who need to update

xidoCohenx committed:
implemented subscribe,

plus navigation to CustomerHome

xidoCohenx committed:
a

xidoCohenx committed:
added the parameter

to update-customer-form navigation

xidoCohenx committed:
changed f  function to getter function

added type validation
and same-customer validation

OrShemesh5020 commented:
@xidoCohenx  on onSubmit function, you forgot to check what if an admin user makes the customer update.
you should start add the following code at row 120 in your last commit :
```
if (this.user.ClientType===ClientType.ADMINISTRATOR){
this.adminService.updateCustomer(this.customerModel).subscribe(()=>{
this.router.navigate(['adminHome']);
});
}
...

xidoCohenx committed:
added admin navigation

and cleaned the code

xidoCohenx committed:
Merge branch 'forms_development' into customer_forms

xidoCohenx committed:
added subscribe to registerCustomer

xidoCohenx committed:
moved initializeFormGroupo into

subscribe

xidoCohenx committed:
Merge branch 'customer_forms' of https://github.com/OrShemesh5020/coupon-app into customer_forms

xidoCohenx committed:
making registerCustomer return

the observable

xidoCohenx committed:
removing parameters

from updateCustomer function
and navigating to updateCustomerForm
